tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py

```diff
@@ -397,23 +397,71 @@ def add_meta_data(net: Element, meta_info: dict):
             SubElement(parameters, 'unset').set('unset_cli_parameters', ', '.join(sorted(meta_info['unset'])))
 
 
+def serialize_node(graph: Graph, node: Node, layers: SubElement, edges: SubElement, unsupported: UnsupportedOps):
+    if node.kind == 'op' and (not node.has('type') or node.type is None):
+        unsupported.add(node)
+        return
+    if not node.has('IE'):
+        return
+    try:
+        serialize_node_attributes(graph, node, node.IE, layers, edges, unsupported)
+    except Error as e:
+        raise Error(str(e).replace('<SUB-ELEMENT>', '{} (id = {})'.format(node.soft_get('name'), node.id))) from e
+
+
 def serialize_network(graph, net_element, unsupported):
     layers = SubElement(net_element, 'layers')
     edges = SubElement(net_element, 'edges')
     if graph is None:
         return
     nodes = sorted(graph.nodes())
+
+    ordered_results = []
+    for output_name in graph.outputs_order:
+        node = graph.get_op_nodes(name=output_name)
+        if len(node) == 0:
+            log.warning("Output node with name {} is not found in graph.".format(output_name))
+            continue
+        node = node[0]
+
+        # In this case Result node has the same name as output tensor
+        if node.soft_get('type') == 'Result':
+            ordered_results.append(node.soft_get('name'))
+            continue
+
+        if len(node.out_nodes()) > 1:
+            log.warning("Node that expected to be output with name {} has more than one output.".format(output_name))
+            continue
+
+        assert len(node.out_nodes()) == 1, "Incorrect graph. Non-Result node with name {} " \
+                                           "has no output node.".format(output_name)
+
+        # After port renumbering port/connection API is not applicable
+        result_node = node.out_node(len(node.in_nodes())).out_node()
```

Comment:
please double check this line. it looks strange.

Reply:
at this stage we know that `len(node.out_nodes()) == 1` so there is the single port. Hence, shouldn't we have just `node.out_node(0).out_node()` or just `node.out_node(0).out_node()`?

Reply:
After port renumbering output port indices start from len(node.out_nodes()), so I can't use 0 output port here. For example if there is 1 input and 1 output port, then input port idx = 0, output port idx = 1. If there are 3 input ports, 2 
output ports, then input ports: 0, 1, 2; output ports: 3, 4. 
So I used node.out_node(len(node.in_nodes())) to get first output port.